
NSCPHD1101  

PhD course in analyses of genotyping and next-generation sequencing data in medical and population genetics 2015/2016

Brief description

The course is a comprehensive introduction to a number of topics and common research tools used in analyses of genetic data. Topics include: genetic drift, natural selection, population structure, Fst, linkage disequilibrium, relatedness, association mapping, GWAS, genotyping data, next-generation sequencing (NGS) data, 1000 genomes project data. The main focus will be on humans, but the methods are applicable for many other species as well. Lectures will be combined with hands-on computer exercises mostly using R and command line based programs.

Learning Outcome

After the course the student will be able to:
- Use population genetic theory to infer basic population genetics characteristics from genetic data, including Fst, population size, relatedness and linkage disequilibrium 
- Be able to infer ancestry and population structure based on genetic data
- Select the optimal strategy for disease mapping based on the available data, the trait of interest, and the population genetics characteristics of the sample population, including taking population structure into account
- Perform disease mapping, including genome-wide association studies
- Interpret and discuss the results of own analyses and results in the scientific literature


Program

Monday – introduction population genetics

    09:00 - 09:10 Registration.
    09:15 - 10:15 Lecture 1: Introduction to basic population genetic terms and concepts. Alleles, 
	genotypes, allele frequencies, the infinite sites model, number of segregating sites, average number of pairwise differences, drift, neutrality, changes in population size and the effect on the coalescent (Fernando)
    10:30 - 12:00 Computer exercise I: e.g. coalescent.dk (Fernando)
    12:00 - 01:00 Lunch 
    01:00 - 02:15 Lecture 2: Introduction to basic population genetic terms and concepts II. The effect of 
	recombination, linkage disequilibrium and measures of this, natural selection, the effect 
	of natural selection on LD and coalescent trees (Anders/maybe Mikkel Schierup)
    02:30 - 04:00 Computer Exercise II: e.g. Coalescent.dk, PLINK, R SNPmatrix, PSMC?
    04:15 - 05:00 Research lecture by Mikkel Schierup?
    05:00 - 07:30 Reception/social mixer

Comments to this day: The idea with this day is to give a basic intro to some of the most important concepts in population genetics (those listed above). Exactly how it is split across the morning and afternoon session is pretty much up to those who will be teaching it. 

Tuesday -  introduction to NGS data

    09:00 - 10:15 Lecture 3: Data processing, mapping, and filtering. (Thorfinn)
    10:30 - 12:00 Computer Exercises III: BWA, SAMtools, Picard, ANGSD, FastQC, and GATK  (Thorfinn)
    12:00 - 01:00 Lunch (on your own)
    01:00 - 02:15 Lecture 4: Allele frequencies, genotypes and SNPs. Maximum likelihood and Bayesian 
	estimation. Genotype likelihoods, estimation of allele frequencies, SNP calling, genotype 
calling, optimization incl. EM algorithm, imputation and phasing (Matteo+Anders for imputation)
    02:30 - 04:00 Computer Exercises IV: ANGSD, R, realSFS, GATK/SAMtools, Beagle (Matteo)
    04:15 - 05:00 Research lecture by Matteo Fumagalli

Comments to this day: The idea with this day is to give a basic intro to NGS data and the tools you use to work with it. 

Wednesday – Demographic and population structure

    09:00 - 10:15 Lecture 5: Demographic inference: FST. Site frequency spectrum. ABBA-BABA (Fernando) 
    10:30 - 12:00 Computer Exercise V: e.g. IM, DADI , ANGSD, and ngsTools (Fernando)
    12:00 - 01:00 Lunch (on your own)
    01:00 - 02:15 Lecture 6:  Inference of population structure (Brenna)
    02:30 - 04:00 Computer Exercises VI: ADMIXTURE, NGSadmix (Brenna)
    04:15 - 05:00 Research lecture by Fernando Racimo

Comments to this day: The idea with this day is to teach the students about inference about demographic history and population structure. 

Thursday – Disease mapping

    09:00 - 10:15 Lecture 7: Association mapping I.  Introduction to single-SNP association mapping, 	exome sequencing, GWAS. (Ida/Anders)

    10:30 - 12:00 Computer Exercises VII: R, PLINK (Ida/Anders)
    12:00 - 01:00 Lunch (on your own)
    01:00 - 02:15 Lecture 8:  Association mapping II.  Whole genome NGS association, correction for 
	population structure/spurious relatedness with PCA/mixed model epistasis, gene-based 
	tests (Anders/Ida)
    02:30 - 04:00 Computer Exercises VIII: Samtools and ANGSD (Anders/Ida).
    04:15 - 05:00 Research lecture by X
Comments to this day: The idea with this day is to teach the students about disease association mapping. 

Friday – Detecting selection and genetic load

    09:00 - 10:15 Lecture 9: Detecting natural selection.  (Matteo) 
    10:30 - 12:00 Computer Exercise IX: e.g. ANGSD/realSFS, PBS 1000 G browser (Matteo)
    12:00 - 01:00 Lunch 
    01:00 - 02:15 Lecture 10: Distribution of genetic load (+ consequences in the context of disease?) (Brenna)
    02:30 - 04:00 Exercises X: ? (Brenna/ skip it and move up the research talk to here and end earlier) 
    04:15 - 05:00 Research lecture by Brenna Henn

Comments to this day: The idea with this day is to teach the students about how one can infer selection and to make them familiar with the concept of genetic load. 



